sql/expression/function/dimension.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+// Dimension is a function that converts a spatial type into WKT format (alias for AsText)
+type Dimension struct {
+	expression.UnaryExpression
+}
+
+var _ sql.FunctionExpression = (*Dimension)(nil)
+
+// NewDimension creates a new point expression.
+func NewDimension(e sql.Expression) sql.Expression {
+	return &Dimension{expression.UnaryExpression{Child: e}}
+}
+
+// FunctionName implements sql.FunctionExpression
+func (p *Dimension) FunctionName() string {
+	return "st_dimension"
+}
+
+// Description implements sql.FunctionExpression
+func (p *Dimension) Description() string {
+	return "returns the dimension of the geometry given."
+}
+
+// IsNullable implements the sql.Expression interface.
+func (p *Dimension) IsNullable() bool {
+	return p.Child.IsNullable()
+}
+
+// Type implements the sql.Expression interface.
+func (p *Dimension) Type() sql.Type {
+	return p.Child.Type()
+}
+
+func (p *Dimension) String() string {
+	return fmt.Sprintf("ST_DIMENSION(%s)", p.Child.String())
+}
+
+// WithChildren implements the Expression interface.
+func (p *Dimension) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 1 {
+		return nil, sql.ErrInvalidChildrenNumber.New(p, len(children), 1)
+	}
+	return NewDimension(children[0]), nil
+}
+
+// Eval implements the sql.Expression interface.
+func (p *Dimension) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	// Evaluate child
+	val, err := p.Child.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	// Return nil if geometry is nil
+	if val == nil {
+		return nil, nil
+	}
+
+	// Expect one of the geometry types
+	switch val.(type) {
```

Comment:
docs claim this can return -1, is that not true?

Reply:
This is output from a couple non-GIS types. I can't think of any that gets `ST_DIMENSION` to return -1.
```
MySQL  localhost:33060+ ssl  test_db  SQL > select st_dimension(1);
ERROR: 3037: Invalid GIS data provided to function st_dimension.
MySQL  localhost:33060+ ssl  test_db  SQL > select st_dimension("testing");
ERROR: 3037: Invalid GIS data provided to function st_dimension.
MySQL  localhost:33060+ ssl  test_db  SQL > select st_dimension(timestamp(0));
ERROR: 3055: Geometry byte string must be little endian.
MySQL  localhost:33060+ ssl  test_db  SQL > select st_dimension(hex("notapoint"));
ERROR: 3055: Geometry byte string must be little endian.
MySQL  localhost:33060+ ssl  test_db  SQL >
 ```